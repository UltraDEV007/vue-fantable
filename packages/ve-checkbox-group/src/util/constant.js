// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
