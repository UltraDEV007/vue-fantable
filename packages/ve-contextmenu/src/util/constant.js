// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
