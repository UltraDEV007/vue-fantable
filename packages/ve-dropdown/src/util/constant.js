// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
