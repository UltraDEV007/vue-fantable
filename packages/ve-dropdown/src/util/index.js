/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
