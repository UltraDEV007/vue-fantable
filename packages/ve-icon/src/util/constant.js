// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
