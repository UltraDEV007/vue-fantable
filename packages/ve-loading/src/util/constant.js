// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
