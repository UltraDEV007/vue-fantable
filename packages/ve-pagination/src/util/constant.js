// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
