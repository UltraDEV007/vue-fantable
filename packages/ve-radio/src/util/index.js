import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
