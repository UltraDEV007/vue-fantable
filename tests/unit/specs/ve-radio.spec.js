import { mount } from "@vue/test-utils";
import veRadio from "@/ve-radio";
import { later } from "../util";

describe("veRadio", () => {
    it("render", () => {
        const wrapper = mount({
            render() {
                const { checked, unChecked } = this;
                return (
                    <div>
                        <div>
                            <veRadio value={checked}>normal</veRadio>
                        </div>
                        <div>
                            <veRadio disabled value={checked}>
                                disabled checked
                            </veRadio>
                        </div>
                        <div>
                            <veRadio disabled value={unChecked}>
                                disabled unChecked
                            </veRadio>
                        </div>
                        <div>
                            <veRadio isControlled isSelected={checked}>
                                controlled
                            </veRadio>
                        </div>
                    </div>
                );
            },
            data() {
                return {
                    checked: true,
                    unChecked: false,
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("value prop", async () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: false,
            },
        });

        expect(wrapper.find(".ve-radio-checked").exists()).toBe(false);

        wrapper.setProps({ isSelected: true });

        await later();
        expect(wrapper.find(".ve-radio-checked").exists()).toBe(false);
    });

    it("disable prop", () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: true,
            },
        });

        expect(wrapper.find(".ve-radio-checked").exists()).toBe(true);
    });

    it("label prop", () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: false,
                label: "test",
            },
        });

        expect(wrapper.find(".ve-radio-label").text()).toContain("test");
    });

    it("isControlled prop", async () => {
        const wrapper = mount(veRadio, {
            propsData: {
                isControlled: true,
                isSelected: false,
            },
        });

        expect(wrapper.find(".ve-radio-checked").exists()).toBe(false);

        wrapper.setProps({ isSelected: true });

        await later();
        expect(wrapper.find(".ve-radio-checked").exists()).toBe(true);
    });

    /*   it("click event", async () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: false,
                label: "test"
            }
        });

        wrapper.trigger("click");

        await later();
        expect(wrapper.find(".ve-radio-checked").exists()).toBe(true);
    }); */

    it("on-radio-change event", async () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: false,
                label: "test",
            },
        });

        wrapper.trigger("click");

        await later();
        expect(wrapper.emitted("on-radio-change").length).toEqual(1);
        expect(wrapper.emitted("on-radio-change")[0]).toEqual([true]);
    });
});
