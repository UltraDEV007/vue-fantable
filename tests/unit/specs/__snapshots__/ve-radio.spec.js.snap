// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veRadio render 1`] = `
<div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">normal</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked ve-radio-disabled"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">disabled checked</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-disabled"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">disabled unChecked</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input type="radio" class="ve-radio-input" value=""><span class="ve-radio-inner"></span></span><span class="ve-radio-label">controlled</span></label></div>
</div>
`;
